Dockerfile

```diff
@@ -0,0 +1,28 @@
+# ---- Base Node ----
+FROM node:11-alpine AS base
+# set working directory
+WORKDIR /app
+# copy project file
+COPY package*.json ./
+
+# ---- Dependencies ----
+FROM base AS dependencies
+# install node packages
+RUN npm set progress=false
+# install ALL node_modules, including 'devDependencies'
+RUN npm install
+
+# ---- Test ----
+## run linters, setup and tests
+FROM dependencies AS test
```

Comment:
I'd say it'd be better to run tests as part of a pre-build action rather than as part of the docker file